cmds/clients/contestcli/transport/transport.go

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package transport
+
+import (
+	"github.com/facebookincubator/contest/pkg/api"
+	"github.com/facebookincubator/contest/pkg/types"
+)
+
+// Transport abstracts different ways of talking to contest server.
+// This interface strictly only uses contest data structures.
+type Transport interface {
+	Version(requestor string) (*api.VersionResponse, error)
+	Start(requestor string, jobDescriptor string) (*api.StartResponse, error)
```

Comment:
What would you use? I used string because it will inevitably be broken up again for transport via http and especially thrift. The client validates it is valid JSON, but that is only done to fail fast, the server has to parse it anyway.

Reply:
[job.JobDescriptor](https://github.com/facebookincubator/contest/blob/11388b8f6e3582b4515ac753840d8d2a58dcc08d/pkg/job/job.go#L20)?

Reply:
Yeah that makes a lot of sense to me, however, I just looked at the internal client and it does this crazy jq-based string manipulation stuff, this likely won't be applicable without some more changes. I think I am going to punt this until after the logic is unified.

Reply:
ok, fair enough